poetapp/frost-api
# refactor: docker and travis config

[PR Process](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-process) - [PR Review Checklist](https://github.com/poetapp/documentation/blob/master/process/pr-review.md#pr-review-checklist)

### Release

Semantic release is enabled for this repository. Make sure you follow the right commit message convention. 
We're using semantic-release's default — [Angular Commit Message Conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines).

## Description of Changes

Resolves #651, #652

### TO TEST
Verify all tests pass by running these commands in the frost-api repo after checking out the branch.
```bash
  $ docker-compose down -v
  $ docker-compose build
  $ docker-compose up -d
  $ docker-compose exec frost-api npm run test:unit
  $ docker-compose exec frost-api npm run test:integration
```
This commit brings frost-api in line with the way docker and travis are
in the node repo. It simplifies the docker setup to one Dockerfile and
one docker-compose.yml file. It also consolidates the three different
start scripts into one.

  - Removes `Dockerfile.build` and updates `Dockerfile`.
  - Removes `docker-compose.development.yml` and updates
    `docker-compose.yml` to define frost-api, poet-node and
    all supporting services needed when developing/testing frost-api.
  - Removes `entrypoint.sh` to simplify container startup.
  - Removes the `start:ci` script in favor of a single start script.
  - Adds mongodb initialization scripts from local-dev repo.
  - Brings in the `docker-build.sh` script from the node repo.
  - Adds `.dockerignore` which fixes a potential bug where a container
    build (linux) has it's `node_modules` directory overwritten by the
    host's copy (possibly macos). Also excludes the 'dist' directory
    which is generated inside the container during the build. Omitting
    just these two directories seems to speed up builds quite a bit.

This commit also brings in the `wait-for-it.sh` script as
`wait-for-it.bash`. The existing wait script already in this repo does
not handle the dual wait for rabbit and mongo that we use in the command
for the poet-node container. Conversely, we still need the original wait
script since consul does not have a bash script installed.  We can try
to address this in a future refactoring.
